pkg/api/universal/conversation.go

```diff
@@ -51,9 +53,29 @@ func (a *Universal) ConverseAlpha1(ctx context.Context, req *runtimev1pb.Convers
 		return nil, err
 	}
 
+	scrubber, err := piiscrubber.NewDefaultScrubber()
+	if err != nil {
+		err := messages.ErrConversationMissingInputs.WithFormat(req.GetName())
+		a.logger.Debug(err)
+		return &runtimev1pb.ConversationAlpha1Response{}, err
+	}
+
 	for _, i := range req.GetInputs() {
+		msg := i.GetMessage()
+
+		if i.GetPiiScrubInput() {
+			scrubbed, sErr := scrubber.ScrubTexts([]string{i.GetMessage()})
+			if sErr != nil {
+				sErr = messages.ErrConversationInvoke.WithFormat(req.GetName(), sErr.Error())
+				a.logger.Debug(sErr)
+				return &runtimev1pb.ConversationAlpha1Response{}, sErr
+			}
+
+			msg = scrubbed[0]
+		}
```

Comment:
We are returning on the first error here rather than doing an `errors.Join` of collected errors on scrubed inputs. That's fine, but is a design decision. Would a user of the API expect all errors of the input be returned, or just the first?

Reply:
I'm a fan of returning early and and not using joined errors as they're harder to parse through. I believe I've seen in batched inferencing (which is what we're mimicking with accepting an input array of messages) early return of errors